sympy/series/limits.py

```diff
@@ -128,7 +128,9 @@ def __new__(cls, e, z, z0, dir="+"):
         z = sympify(z)
         z0 = sympify(z0)
 
-        if z0 is S.Infinity:
+        if z0 is S.ComplexInfinity:
+            raise NotImplementedError()
```

Comment:
Not the right place for this. The **construction** of an inert Limit object should not raise a NotImplementedError. The user isn't asking SymPy to compute the limit. The Limit object may be needed to express some logic in a formula, to be returned to a user, or to be processed by a user-written complex-limit-evaluating function. 

What is not implemented is the __evaluation__ of such limits, and that is in the `doit` method.